### ISSUE TITLE
Not able to login in Sugar Environment on ubuntu 18.04

### ISSUE THREAD

**Issue by aditya113141** [sugarlabs/sugar#938]
I installed Sugar according to the instructions provided here:-  https://github.com/sugarlabs/sugar/blob/master/docs/ubuntu.md

It ran well on my pc initially but now When I try to login to the environment, I get  redirected  to the login window as shown in the video.

https://drive.google.com/file/d/1jNgNx3jse7GsHwiypEHVltFWc5FSheHE/view?usp=sharing

**Comment by srevinsaju** [comment]
> We request you to not use google drive to post your Videos. GitHub now natively supports sharing Videos.

**Comment by quozl** [comment]
You were not redirected.  What you see is Sugar failing to start, and the greeter recognising that the desktop environment did not start.

Possibilities are too many.  Need more focus.

Which release of Ubuntu?

What is the contents of the file `.sugar/default/logs/shell.log`?

**Comment by srevinsaju** [comment]
> Which release of Ubuntu?

Ubuntu 18.04 Bionic, but mentioned in the Issue title only.

**Comment by quozl** [comment]
Thanks, I missed that.  According to my [earlier tests](https://github.com/sugarlabs/sugar/blob/master/docs/ubuntu.md#ubuntu-1804-bionic-beaver-and-1810-cosmic-cuttlefish) it did work once upon a time.  Presumably the root causes are changes to dependencies made by unattended upgrades, or a difference in instructions followed.  Still too many possibilities; we need the shell.log file.

**Comment by ThomasJRyan** [comment]
Sorry to bring up something so old, but I'm also having the same issue, only instead of Ubuntu this is happening with both a fresh install of Fedora 35 and a Sugar on a Stick USB.

I tried looking for the `shell.log` at `.sugar/default/logs/shell.log` but it doesn't exist. The only thing I could find was `.sugar/default/debug`.

If you can direct me to where I can find more information I would be happy to provide it

**Comment by ThomasJRyan** [comment]
Apologies for the double post. But after looking around I found that the `decorator` package in python wasn't being installed. I was able to resolve this issue by running
```
python3 -m pip install decorator
```

**Comment by aperezbios** [comment]
@ThomasJRyan thank you for the report. We appreciate it. I'm able to replicate it. It sounds like a missing dependency. We'll work on fixing it, and, for the time being, documenting it. Presumably, you installed it via the console CLI?

**Comment by aperezbios** [comment]
The Fedora-packaged version of the 'decorator' python3 package is named python3-decorator, and can be installed by pressing ctrl-alt-F2, logging in as 'liveuser' (there's no password), and then running 'dnf install -y python3-decorator'

@chimosky , when you have a moment, we need to fix the F35 Sugar (GTK3?) package, to explicitly add python3-decorator as a dependency.

**Comment by quozl** [comment]
Yes, and sugar-toolkit-gtk3:src/sugar3/logger.py has the import of decorator.  Looking further, I'm not sure if it is truly necessary, or even if the code path is used.  I thought Python's exception reporting mechanisms had improved a lot since the code was written.

**Comment by chimosky** [comment]
> Apologies for the double post. But after looking around I found that the `decorator` package in python wasn't being installed. I was able to resolve this issue by running
> 
> ```
> python3 -m pip install decorator
> ```

Thanks for reporting this, can you share your process of looking around?

**Comment by ThomasJRyan** [comment]
> > Apologies for the double post. But after looking around I found that the `decorator` package in python wasn't being installed. I was able to resolve this issue by running
> > ```
> > python3 -m pip install decorator
> > ```
> 
> Thanks for reporting this, can you share your process of looking around?

It wasn't anything special or exciting. After being unable to load into the DE I dropped into the shell instead using the method @aperezbios mentioned. On a whim I tried just running `sugar` and the error popped up. So I tried installing the missing package and it fixed my issues

**Comment by chimosky** [comment]
> @chimosky , when you have a moment, we need to fix the F35 Sugar (GTK3?) package, to explicitly add python3-decorator as a dependency.

Done.

I'll still test why the problem is with `python3-decorator` as I don't see why that should've been a problem.

**Comment by chimosky** [comment]
> It wasn't anything special or exciting. After being unable to load into the DE I dropped into the shell instead using the method @aperezbios mentioned. On a whim I tried just running `sugar` and the error popped up. So I tried installing the missing package and it fixed my issues

Thanks, been having issues with switching to cli with that shortcut, don't know if that's a problem with F34 as I've been having some issues since upgrading.

**Comment by ThomasJRyan** [comment]
I suspect there might be another issue with downloading XO apps from "Browse" that could potentially be related to this, as when I tried last night nothing would install.

I say it might be related because it has the same sort of error where it starts and then abruptly stops, so there could be another dependency issue someone floating around.

**Comment by aperezbios** [comment]
@ThomasJRyan without knowing which specific activities you were attempting to install/use, we can't really diagnose the reason. The most likely reason is probably that the activity you're trying to install/use is a python 2 activity, and Fedora has dropped Python 2 from their default installs, since it is end-of-lifed.
